File: SparseMatrix.py
#4.Write a program to check whether a matrix is sparse or not.

def is_sparse(matrix):
    rows = len(matrix)
    cols = len(matrix[0]) if rows > 0 else 0
    total = rows * cols
    zero_count = sum(row.count(0) for row in matrix)

    return zero_count > total // 2

# Example matrix
matrix = [
    [0, 0, 3],
    [0, 0, 6],
    [7, 0, 0]
]

print("Matrix:")
for row in matrix:
    print(row)

if is_sparse(matrix):
    print("The matrix is a sparse matrix.")
else:
    print("The matrix is not a sparse matrix.")

File: SparseMatrix.c
//Sparse Matrix

#include <stdio.h>

int main() {
    int rows, cols,i,j;
    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols];
    int zeroCount = 0, total = rows * cols;

    printf("Enter the elements of the matrix:\n");
    for (i = 0; i < rows; i++)
        for ( j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
            if (matrix[i][j] == 0)
                zeroCount++;
        }

    if (zeroCount > total / 2)
        printf("The matrix is a sparse matrix.\n");
    else
        printf("The matrix is not a sparse matrix.\n");

    return 0;
}

